examples/axum-spa/src/main.rs

```diff
@@ -0,0 +1,72 @@
+use axum::{
+    body::{boxed, Full},
+    handler::Handler,
+    http::{header, StatusCode, Uri},
+    response::{IntoResponse, Response},
+    routing::Router,
+};
+use rust_embed::RustEmbed;
+use std::net::SocketAddr;
+
+static INDEX_HTML: &str = "index.html";
+
+#[derive(RustEmbed)]
+#[folder = "assets/"]
+struct Assets;
+
+#[tokio::main]
+async fn main() {
+    let app = Router::new().fallback(static_handler.into_service());
+
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+    println!("listening on {}", addr);
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
+}
+
+async fn static_handler(uri: Uri) -> impl IntoResponse {
+    let path = uri.path().trim_start_matches('/');
+
+    if path.is_empty() || path == INDEX_HTML {
+        return index_html().await.into_response();
+    }
+
+    match Assets::get(path.as_str()) {
```

Comment:
`path` is already a `&str`, so there's a compile time error:
```
error[E0599]: no method named `as_str` found for reference `&str` in the current scope
    --> examples/axum-spa/main.rs:33:26
     |
33   |   match Assets::get(path.as_str()) {
     |                          ^^^^^^
     |
help: remove this method call
     |
33   -   match Assets::get(path.as_str()) {
33   +   match Assets::get(path) {
     | 
```

```suggestion
    match Assets::get(path) {
```

Reply:
Done.